server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -259,6 +260,11 @@ func (w *writeState) Write(buf []byte) error {
 }
 
 func (w *writeState) Close() error {
+	if w.closed {
+		return nil
+	}
+	w.closed = true
+
```

Comment:
could you nil out w.decompressorCloser after closing it? and remove this block?

Reply:
lol, I had that before, but was on the fence. Changed it back

Reply:
it's somewhat idiomatic in go to do that afaik. Don't have a source handy tho, lmk if you find one.